### PR TITLE
Golangci lint

### PR DESCRIPTION
**Pull request by dertseha**
Hello there!
A fine library that you have here, and since you were mentioning linter issues in #11 , I'd like to provide you with a starter package :)

Things done:
* Added config file for golangci-lint -- With that, you can start integration at https://golangci.com and perhaps add a badge if you want.
* Replaced all block comments with single-line comments
* Added comments where they were missing
* Renamed `Cancelled` to `ErrCancelled`, and added a `Cancelled` forwarder, marked as "Deprecated" (which gets striked-through in IDEs, such as GoLand)
* go fmt all over
* Split up one long line (lll)
* renamed `cocoa/dlg.go` to `cocoa/dlg_darwin.go`

### CODE REVIEW

**Comment by sqweek** [comment]
Thanks for this! Does dlg.go -> dlg_darwin.go have any practical benefit? The entire cocoa/ package is of course OSX specific and it shouldn't be imported elsewhere.

File: dlgs.go [hunk 1]
```diff
@@ -1,89 +1,95 @@
...
 )
 
-/* Cancelled is an error returned when a user cancels/closes a dialog. */
-var Cancelled = errors.New("Cancelled")
+// ErrCancelled is an error returned when a user cancels/closes a dialog.
+var ErrCancelled = errors.New("Cancelled")
 
+// Cancelled refers to ErrCancelled.
+// Deprecated: Use ErrCancelled instead.
+var Cancelled = ErrCancelled
```
Comment (sqweek) [inline dlgs.go]: Nice way to retain backwards compatibility :+1:

File: example/wde/main.go [hunk 2]
```diff
@@ -2,7 +2,7 @@ package main
 
 import (
 	"fmt"
-	"github.com/skelterjohn/go.wde"
+	"github.com/skelterjohn/go.wde" // nolint: goimports
```
Comment (sqweek) [inline example/wde/main.go]: I don't love the magic comment here, it's a perfectly normal import?

File: example/wde/main.go [hunk 3]
```diff
@@ -17,8 +17,7 @@ func events(events <-chan interface{}) {
 	for ei := range events {
 		switch e := ei.(type) {
 		case wde.MouseUpEvent:
-			switch e.Which {
-			case wde.LeftButton:
+			if e.Which == wde.LeftButton {
```
Comment (sqweek) [inline example/wde/main.go]: I'd prefer to leave this as a switch -- even though it's pointless here, there's an advantage to having this example code mimic the usual event-loop code structure.

File: example/wde/main.go (hunk 2 above)
Comment (dertseha) [reply example/wde/main.go]: One would think, yes - sadly, the linter has a standing issue to trip up with special characters in package names (such as `.` or `-`) and then complains about wrong import sorting. :/

File: example/wde/main.go (hunk 3 above)
Comment (dertseha) [reply example/wde/main.go]: Well, not according to gocritic linter ;)
Restoring the switch would require a `nolint` tag - acceptable?

**Comment by dertseha** [comment]
Thank you for your review!

> Thanks for this! Does dlg.go -> dlg_darwin.go have any practical benefit? The entire cocoa/ package is of course OSX specific and it shouldn't be imported elsewhere.

running tools like "go test ./..." or similar that need to compile stuff will trip up if the environment is not darwin. Hence I made the dependency explicit.

**Comment by dertseha** [comment]
To avoid a state in which we might be waiting on each other, I'm asking - Is there anything I can or should do? See also discussion points above.

**Comment by sqweek** [comment]
Sorry, this is blocked on me bc my personal life is a mess and I'm struggling with focus in my spare time. I don't love the linter comments creeping in but please add the `nolint` on the example/wde/main.go switch statement and I'll squash and land.

**Comment by dertseha** [comment]
No need to rush - now I was the one occupied.
I've updated the `main.go` file, using version 1.16.0 of golangci-lint .

**Comment by sqweek** [comment]
Awesome, I've finally landed this -- thanks so much for your time!

**Pull request closed without merging**